galaxy_ng/app/api/ui/viewsets/execution_environment.py

```diff
@@ -346,6 +351,34 @@ class ContainerRegistryRemoteViewSet(api_base.ModelViewSet):
     permission_classes = [access_policy.ContainerRegistryRemoteAccessPolicy]
     filterset_class = ContainerRegistryRemoteFilter
 
+    @extend_schema(
+        description="Trigger an immediate delete task",
+    )
+    @transaction.atomic
+    def destroy(self, request: Request, *args, **kwargs) -> Response:
+        """Delete a container Registry.
+
+        Delete the selected ContainerRegistryRemote object from the DB
+        For each Container Remote that's connected to the registry:
+        Delete the association between the remote and the registry
+        Clear the remote's connection information.
+
+        This will make it so that container remotes that are associate with a deleted registry will
+        still be available and can be re-added to another registry, but they won't be able to
+        be re-synced.
+
+        return: Response(status=204)
+        """
+
+        registry = self.get_object()
+
+        for remote_association in registry.containerregistryrepos_set.all():
```

Comment:
This might not be necessary. The model is already set to cascade if the registry is deleted: https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/models/container.py#L84

https://stackoverflow.com/questions/38388423/what-does-on-delete-do-on-django-models

Reply:
You may not have to create a destroy method at all for this. The one  from the parent viewset might be enough.